clippy_lints/src/matches/mod.rs

```diff
@@ -871,7 +871,7 @@ declare_clippy_lint! {
     /// ```no_run
     /// fn foo(fail: bool) -> Result<i32, String> {
     ///     if fail {
-    ///       Err("failed")?;
+    ///       Err("failed".into())?;
```

Comment:
Can you elaborate on what this fixes? The `?` operator calls `.into()` by itself so this seems like it should not be needed?

Reply:
Ok, I didn't realize that. But I still consider it is confusing. Maybe we should make the difference between examples minimal?